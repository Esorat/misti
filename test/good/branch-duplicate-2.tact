fun test(): Int {
    let a: Int = 42;
    return (a == 42) ? a + 43 : a + 43; // bad
}
