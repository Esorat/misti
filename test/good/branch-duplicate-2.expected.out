[HIGH] BranchDuplicate: Duplicated code in conditional branches is detected
test/good/branch-duplicate-2.tact:3:12:
  2 |     let a: Int = 42;
> 3 |     return (a == 42) ? a + 43 : a + 43; // bad
                 ^
  4 | }
Help: Identical code in both branches detected. Refactor to eliminate redundancy.
See: https://nowarp.io/tools/misti/docs/detectors/BranchDuplicate